src/server_state.cpp

```diff
@@ -691,8 +691,13 @@ void wsrep::server_state::sst_sent(const wsrep::gtid& gtid, int error)
 
     wsrep::unique_lock<wsrep::mutex> lock(mutex_);
     // v26 API does not have JOINED event, so in anticipation of SYNCED
-    // we must do it here.
-    state(lock, s_joined);
+    // we must do it here. Do not modify the state if donor lost the
+    // donor state e.g. due to cluster partitioning.
+    if (state(lock) == s_donor)
+    {
+        state(lock, s_joined);
+    }
+
```

Comment:
How about adding the same fix to sart_sst() above?

Reply:
Good catch, will take a look.